### PR TITLE
__main__.py: Check docker image string

### PR DESCRIPTION
**Pull request by PrajwalM2212**
This PR tests that the docker image string provided to tern is of image:tag or image@digest_type:digest format

Closes https://github.com/vmware/tern/issues/519

Signed-off-by: PrajwalM2212 <prajwalmmath@gmail.com>

### CODE REVIEW

**Comment by PrajwalM2212** [comment]
The regex used here is meant to confirm `image@digest_type:digest` and `image:tag` format.

File: tern/__main__.py [hunk 1]
```diff
@@ -67,6 +68,14 @@ def create_top_dir():
...
 
 
+def check_docker_image_string(image_str: str):
+    tag_format = r'.+:.+'
+    digest_format = r'.+@.+:.+'
+    if re.match(tag_format, image_str) or re.match(digest_format, image_str):
+        return True
+    return False
+
+
```
Comment (nishakm) [inline tern/__main__.py]: Good job! Couple of organization nits:
1. This seems to be the accepted format to reference any container image on a registry. Could you change the name of this function to `check_image_string`?
2. Can you move this function to `tern/utils/general.py`?

File: tern/__main__.py [hunk 2]
```diff
@@ -86,6 +95,14 @@ def do_main(args):
         if args.dockerfile:
             run.execute_dockerfile(args)
         if args.docker_image:
+            # Check if the image is of image:tag
+            # or image@digest_type:digest format
+            if not check_docker_image_string(args.docker_image):
+                sys.stderr.write('Error running Tern\n'
+                                 'Please provide docker image '
+                                 'string in image:tag or '
+                                 'image@digest_type:digest format\n')
```
Comment (nishakm) [inline tern/__main__.py]: Perfect!

File: tests/test_main.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import unittest
+from tern.__main__ import check_docker_image_string
+
+
+class TestMain(unittest.TestCase):
+
+    def test_docker_image_str(self):
```
Comment (nishakm) [inline tests/test_main.py]: 1. Can you move this test to `tests/test_util_general.py`?
2. Can you keep unittest's convention for naming tests? `testImageString`?

**Comment by nishakm** [comment]
> The regex used here is meant to confirm `image@digest_type:digest` and `image:tag` format. It is not for ascertaining whether the image string is according the image naming guidelines.

There are no real naming guidelines currently. However, we may have to add a detection for transport (`docker:// vs oci://`), but we can worry about that later.

**Comment by nishakm** [comment]
Looks good. Merging. Thanks a lot @PrajwalM2212!

**Pull request merged**